src/GI.h

```diff
@@ -41,8 +45,14 @@ template <class Ptcl> class GI : public Problem<Ptcl>{
 		const PS::F64 impCoreMass = impMass * coreFracMass;
 		const PS::F64 impCoreRadi = impRadi * coreFracRadi;
 
+		
+
 		const double offset = 5.0 * UnitRadi;
-		const PS::F64 dx = 1.0 / 39;
+		// the following line predicts the number of grid points in one direction
+		const int  gridpoint = int(2.0/pow(4.18*1.1/Nptcl,0.333));
```

Comment:
These are a lot of "magic numbers". It would be easier to understand with the following:
```suggestion
                // Put a comment here what this formula is, in particular: why is it 2 / (V^1/3)? And why is there a factor of 1.1 in there? 
		const unsigned int n_gridpoints = static_cast<unsigned int>(2.0/pow(4.0/3.0*math::pi*1.1/Nptcl,1.0/3.0));
```

Reply:
Thank you - I added some comments on this

Reply:
Thanks for the documentation. Could you still change `4.18` to `4.0/3.0 * math::pi`? I know it is just a guess, but no need to obscure the formula behind the guess.